ui/lib/sync/addon/routes/secrets/overview.ts

```diff
@@ -38,7 +41,7 @@ export default class SyncSecretsOverviewRoute extends Route {
   }
 
   redirect() {
-    if (!this.flags.showSecretsSync) {
+    if (!this.version.hasSecretsSync) {
```

Comment:
One small change in functionality here is this no longer _also_  includes a check for nav permission. We usually don't do permissions checks in redirects and instead hide/show links so users can't navigate there in the first place.

If folks disagree, I can add it back

Reply:
The idea was that if a user only has the ability to Activate secret sync but they don't have access to secret sync functionality, allow them to activate and then redirect them to dashboard.

Reply:
The previous check wasn't doing anything to check for ability to activate, so it sounds like we need to add the activation endpoint here: https://github.com/hashicorp/vault/blob/main/ui/app/services/permissions.js#L52-L57